### ISSUE TITLE
Feature Request - Statistics on users who gave a given tag and of the tool used

### ISSUE THREAD

**Issue by Codain** [taginfo/taginfo#221]
When doing some reviews on Taginfo to detect common errors (i.e. spelling mistakes, tag which should be unique but is not...) I often wonder  if this is a sum of little errors or if it's due to a user/a tool. If one user has introduced a lot of mistakes, one can contact him and explain him what he thinks is wrong in order to improve future changeless (and fix errors). Or if a tool introduce a mistaken one can report a bug to the developpers in order to fix it.

The idea is, for a given key=value combination, to display in a dedicated tab:

- how many have been assigned by each users (i.e. a table with a counter and a username)
- how many have been introduced via each tool (i.e. a table with a counter and a tool name)

Algorithm proposal:
For each occurrence of the tag, find the changeset which introduced this tag in the object and extract from the changeset the user name and the tool name.

**Comment by joto** [comment]
I agree that this would be useful information. It would mean going through the full history which is a lot of effort. I leave the issue open, but I wouldn't expect this to be implemented soon. (If somebody else wants to work on this, please do.)

**Comment by Codain** [comment]
@joto Maybe this can be a good candidate for next GSoC? If so it's probably better you propose it because a mentor would be needed: https://wiki.openstreetmap.org/wiki/Google_Summer_of_Code/2021/Project_ideas . This one or another issue you have in mind if Taginfo is open to GSoC contributions (time to spend...).

For tools, I think it would be nice to play with regex to separate software name from its version in 'created_by' whenever possible (for instance JOSM and iD both use a stable in time schema). This can bring an additional level of detail to track down what happened.
